### PR TITLE
Keras 3 compatibility updates

### PR DESCRIPTION
**Pull request by divyashreepathihalli**
Uses tf.random.Generator instead of keras.backend.RandomGenerator
removed legacy_make_seed  

Tests disabled - https://github.com/keras-team/keras-cv/issues/2167
- all 3d preprocessing tests - has legacy baserandomlayer
- rand_augment_test - need to investigate
- random_shear_test - need to investigate
- repeated_augmentation_test - need to investigate
- resizing_test - has ragged tensor tests and is failing on jax and torch

### CODE REVIEW

File: benchmarks/vectorized_mosaic.py [hunk 1]
```diff
@@ -101,7 +101,9 @@ def _batch_augment(self, inputs):
             minval=0,
             maxval=batch_size,
             dtype=tf.int32,
-            seed=self._random_generator.make_legacy_seed(),
+            seed=self._random_generator.uniform(
```
Comment (sampathweb) [inline benchmarks/vectorized_mosaic.py:104]: Are we changing the functionality here when we use random uniform to generate seed?

File: keras_cv/layers/preprocessing/fourier_mix.py [hunk 2]
```diff
@@ -54,10 +54,12 @@ def __init__(self, alpha=0.5, decay_power=3, seed=None, **kwargs):
 
     def _sample_from_beta(self, alpha, beta, shape):
         sample_alpha = tf.random.gamma(
-            shape, alpha=alpha, seed=self._random_generator.make_legacy_seed()
+            shape,
+            alpha=alpha,
```
Comment (sampathweb) [inline keras_cv/layers/preprocessing/fourier_mix.py:58]: If we remove the seed, are we expecting same results like before?

File: keras_cv/layers/preprocessing/random_color_jitter_test.py [hunk 3]
```diff
@@ -22,9 +21,9 @@
 class RandomColorJitterTest(TestCase):
     # Test 1: Check input and output shape. It should match.
     def test_return_shapes(self):
-        batch_input = np.ones((2, 512, 512, 3))
-        non_square_batch_input = np.ones((2, 1024, 512, 3))
-        unbatch_input = np.ones((512, 512, 3))
+        batch_input = tf.ones((2, 512, 512, 3))
```
Comment (sampathweb) [inline keras_cv/layers/preprocessing/random_color_jitter_test.py:24]: why do we need to change them to `tf.`?  I think `np.` should also work since we allow for numpy args with the variable set in base layer

Comment (sampathweb) [reply keras_cv/layers/preprocessing/random_color_jitter_test.py:24]: Same in many tests

File: keras_cv/layers/preprocessing/random_crop_test.py [hunk 4]
```diff
@@ -31,7 +32,7 @@ class RandomCropTest(TestCase):
         ("random_crop_full_width", 3, 8),
     )
     def test_output_shape(self, expected_height, expected_width):
-        np.random.seed(1337)
+        tf.random.set_seed(1337)
```
Comment (sampathweb) [inline keras_cv/layers/preprocessing/random_crop_test.py:35]: Same here.  Why do we need to use `tf` instead of `np.`?

Comment (divyashreepathihalli) [reply keras_cv/layers/preprocessing/random_crop_test.py]: done

File: keras_cv/layers/preprocessing/random_color_jitter_test.py (hunk 3 above)
Comment (divyashreepathihalli) [reply keras_cv/layers/preprocessing/random_color_jitter_test.py]: done

**Comment by sampathweb** [comment]
Replaced by `keras-3-updates` PR
